src/info/langs/language.rs

```diff
@@ -101,10 +109,18 @@ impl std::fmt::Display for LanguagesInfo {
     }
 }
 
+#[derive(Debug, PartialEq)]
+struct LanguageDisplayData {
+    language: String,
+    percentage: f64,
+    circle_color: DynColors,
+    icon: &'static str,
+}
+
 fn prepare_languages(
```

Comment:
You should add a unit test on this method to cover the case when `languages_info.nerd_fonts` is true

Reply:
To make this a bit easier to test, I'd recommend writing a method that's implemented on either `Language` or `LanguagesInfo`. This method would resolve the icon or fall back to the circle when it's either disabled or not supported.

For example:
```rust
// on language
impl Language {
    fn get_chip(&self, use_nerd_fonts: bool) -> &'static str {
        if !use_nerd_fonts {
            return DEFAULT_ICON;
        }
        self.get_icon().unwrap_or(DEFAULT_ICON)
    }
}
```

And if you were to implement it on `LanguagesInfo`, it would take `language` as an argument and use `self.nerd_fonts`.

You don't need to use this exact implementation, but this is an example of how you could organize code to make it easier to test.

Reply:
I wrote a test for this case before I saw your suggestion.

It's identical to the one before it except for the value of `nerd_fonts`, and the expected result.

Is it worth changing it?

```rust
    fn test_prepare_languages_with_nerd_fonts() {
        let languages_info = LanguagesInfo {
            languages_with_percentage: vec![
                LanguageWithPercentage {
                    language: Language::Go,
                    percentage: 40_f64,
                },
                LanguageWithPercentage {
                    language: Language::Erlang,
                    percentage: 30_f64,
                },
                LanguageWithPercentage {
                    language: Language::Java,
                    percentage: 20_f64,
                },
                LanguageWithPercentage {
                    language: Language::Rust,
                    percentage: 10_f64,
                },
            ],
            true_color: false,
            number_of_languages_to_display: 2,
            info_color: DynColors::Ansi(AnsiColors::White),
            nerd_fonts: true,
        };

        let color_palette = [
            DynColors::Ansi(AnsiColors::Red),
            DynColors::Ansi(AnsiColors::Green),
        ];

        let result = prepare_languages(&languages_info, &color_palette);

        let expected_result = vec![
            LanguageDisplayData {
                language: Language::Go.to_string(),
                percentage: 40_f64,
                chip_color: DynColors::Ansi(AnsiColors::Red),
                icon: Language::Go.get_icon().unwrap(),
            },
            LanguageDisplayData {
                language: Language::Erlang.to_string(),
                percentage: 30_f64,
                chip_color: DynColors::Ansi(AnsiColors::Green),
                icon: Language::Erlang.get_icon().unwrap(),
            },
            LanguageDisplayData {
                language: "Other".to_string(),
                percentage: 30_f64,
                chip_color: DynColors::Ansi(AnsiColors::White),
                icon: DEFAULT_ICON,
            },
        ];

        assert_eq!(result, expected_result);
    }
```

Reply:
I applied @spenserblack's suggestion in 43023ed

Should be good to go now